lib/Doctrine/Migrations/Version/Executor.php

```diff
@@ -345,10 +345,11 @@ private function executeVersionExecutionResult(
 
             $this->outputSqlQuery($key, $query);
 
+            // executeQuery() must be used here because $query might return a result set, for instance REPAIR does
```

Comment:
then, to be sure to be compatible with `PrimaryReadReplicaConnection` (which allows `executeQuery` to run on the read replicas), we should add a check to call `ensureConnectedToPrimary` when the connection is a `PrimaryReadReplicaConnection` (for transactional migrations, there is a good chance to be connected on the primary due to starting the transaction, but it is possible to disable all usages transactions in doctrine/migrations AFAIK)

Reply:
That wouldn't qualify as a bugfix, so in 3.1.x, right? That being said, your remark makes me notice that we have `instanceof MasterSlaveConnection` that should be replaced with `instanceof PrimaryReadReplicaConnection`

Reply:
Attempt at a fix: https://github.com/doctrine/migrations/pull/1096